src/components/Header/Header.jsx

```diff
@@ -1,11 +1,20 @@
 import React from "react";
 import "./Header.css";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
 import { BsCart, BsHeart, BsPersonCircle } from "react-icons/bs";
-import { useUser } from "../../context";
+import { useUser, useProducts } from "../../context";
 
 export const Header = () => {
   const { user } = useUser();
+  const { dispatch } = useProducts();
+  const navigator = useNavigate();
+
+  const onSearch = (e) => {
+   if(window.location.pathname !== '/products')
+      navigator("/products");
+    return dispatch({ type: "PRODUCTS_SEARCH", payload: e.target.value });
```

Comment:
You don't need to return `dispatch()` function

Reply:
done